sidecar/app/routes/stop.py

```diff
@@ -15,13 +14,16 @@
 @router.post("/finish/{query_id}")
 def finish(
     query_id: str,
+    request: Request,
 ):
-    query = Query.get_from_query_id(query_id)
+    query_manager = request.app.state.QUERY_MANAGER
+    query = query_manager.get_from_query_id(Query, query_id)
     if query is None:
-        return {"message": "Query not found", "query_id": query_id}
-    logger.info(f"{query=}")
+        raise HTTPException(status_code=404, detail="Query not found")
```

Comment:
these null checks and rethrowing 404 are really repetitive. I think we should wrap query manager into a class in the HTTP layer that will do it in one place, so every handler does not need to do it on its own

Reply:
updated it. I just used a wrapper function - we can wrap query manager if there end up being more things like this.